app/models/config.rb

```diff
@@ -70,4 +95,100 @@ def self.start_day
     start ||= "monday"
     start.downcase.to_sym
   end
+
+  private
+    ### Pre-validation cleanup
+
+    def clean_config_value
+      # do nothing if empty
+      return if config_key.nil? || options.last.nil?
+
+      clean_f = CLEAN_PRE_VALIDATION[config_key.to_sym]
+
+      # no clean function, return
+      return if clean_f.nil?
+
+      method(clean_f).()
+    end
+
+    def clean_url
+      url = options.last
+
+      # don't try to clean empty url
+      return if url.blank?
+
+      # don't have to do anything, already https
+      return if url.start_with? 'https://'
+
+      # convert http or // to https://
+      if url.start_with? /(http:)?\/\//
```

Comment:
Would this make more sense in `UriService`? This is quite a bit of very particular url handling, wonder if it makes sense to extract it into our service class.